modules/android/pages/vector-search-api-reference.adoc

```diff
@@ -0,0 +1,286 @@
+
+   /* The max number of vectors used for training the index. The default 
+      value is 256 times number of centroids. The number must be more than zero 
+      and not less than minTrainingSize. */
+   public var maxTrainingSize: Unit = 256 * centroids
+
+   /* VectorIndexConfiguration Constructor.
+      @param expression The SQL++ expression returning a vector which is an 
+                        array of numbers.
+      @param dimensions The number of dimensions of the vectors to be indexed.  
+                        The vectors that do not have the same dimensions 
+                        specified in the config will not be indexed. The
+                        dimensions must be between 2 and 2048 inclusively.
+      @param centroids  The number of centroids which is the number buckets 
+                        to partition the vectors in the index. The number of       
+                        centroids will be based on the expected number of 
+                        vectors to be indexed; one suggested rule is to use 
+                        the square root of the number of vectors. The
+                        centroids must be between 1 and 64000 inclusively. */
+   public init(expression: String, dimensions: UInt, centroids: UInt) { }
+}
+
+----
+
+IMPORTANT: The beta release currently only supports the format of vectors as an array of numbers.
+
+=== Default Value Constants
+
+[source, c]
+----
+
+public class VectorIndexConfiguration {
+  /* Scalar Quantizer */
+  public static let defaultEncoding = VectorEncoding.scalarQuantizer()
+
+  /* Squared Euclidean Distance */
+  public static let defaultMetric = VectorMetric.euclidean   
+}
+
+----
+
+=== Extension Path Configuration
+
+NOTE: The extension configuration is only necessary for the CBL-iOS MacOS and CBL-C platforms as dynamic extension libraries may be at arbitrary locations.
+
+[source, c]
+----
+
+public class CouchbaseLite {
+  /* Registers a directory path to load extension libraries from. Must 
+     be called before using CouchbaseLite but only when using extensions */
+  public static func setExtensionPath(extensionPath: String)
+}
+
+----
+
+== Vector SQL++ Functions
+
+Couchbase Lite currently supports two SQL++ functions, `vector_match()` and `vector_distance()`.
+
+=== Vector Match
+
+==== `vector_match(vectorIndexIdentifier, targetVectorExpr, [limit = 3])`
+
+[cols = "3*"]
+|===
+|Parameter |Is Required |Description
+
+|vectorIndexIdentifier
+|Required
+|The name of the vector index to perform the vector search on.
+|targetVectorExpr
+|Required
+|The target vector expression that returns a vector in the form of an array of numbers.
+|limit
+|Optional
+|The limit number of the returned matched results.
+The maximum number allowed is 10000. An error will be returned when creating a query with a limit greater than 10000.
+
+|===
+
+This function performs vector search against a specific vector index identifier for the specified vector expression.
```

Comment:
```suggestion
This function performs a vector search against a specific vector index identifier for the specified vector expression.
```
Do we want to be capitalizing Vector Search at all in this description? Try to keep consistent.